### PR TITLE
[core] Remove ngraph/opsets API

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - Remove `ngraph/opsets` API.

### Tickets:
 - [CVS-130104](https://jira.devtools.intel.com/browse/CVS-130104)

### CODE REVIEW

File: src/core/include/openvino/opsets/opset.hpp [hunk 1]
```diff
@@ -10,7 +10,18 @@
 #include <set>
 #include <utility>
 
+#ifndef IN_OV_COMPONENT
+#    define IN_OV_COMPONENT
+#    define WAS_OV_LIBRARY_DEFINED_SERIALIZE
+#endif
+
 #include "ngraph/factory.hpp"
```
Comment (olpipi) [inline src/core/include/openvino/opsets/opset.hpp:18]: As I can see, we want to get rid of ngraph factory in scope of CVS-130103
Maybe it is reasonable to unite these two PRs?

Comment (olpipi) [inline src/core/include/openvino/opsets/opset.hpp:18]: This is probably conflict with https://github.com/openvinotoolkit/openvino/pull/22371

Comment (praasz) [reply src/core/include/openvino/opsets/opset.hpp:18]: Will sync depends which one merge first. I think factory removal will be first so then remove this include.
Without it build will fail.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]

**Comment by praasz** [comment]
Depends on #22371

**Review by t-jankowski — APPROVED** [review]

**Pull request merged**